articles/140_topic_and_service_name_mapping.md

```diff
@@ -219,13 +219,83 @@ Since ROS 2 topic and service names are expanded to fully qualified names, any b
+However, aliasing is easily douable with it.
+That is, having an original topic such as `cmd_vel` in a namespace `robot1`, we can easily duplicate that topic as `robot2/cmd_vel`.
+
+#### Hierarchy with DDS Partitions
+
+Given that DDS Partitions are not hierarchical, we have to setup the hierarchy on top.
+For this, we use the forward slash `/` explicitely in each partition field.
+If we want to replicate a hierarchy of `warehouse`, `robot1`, `camera_left`, and finally the basename `image_raw`, the first entry of the partitions would look like:
+
+```
+warehouse/robot1/camera_left/image_raw
+```
+
+#### Remapping with DDS Partitions
+
+We have to differentiate remapping from aliasing at this point.
+As described before, aliasing implies a duplication a topic.
+Remapping, in contrast, alters an existing topic and thus remaps one topic into another.
+Given the nature of DDS Partitions, aliasing is natively supported by adding a second entry in the partition field.
+Remapping means, we have to change one existing partition field.
+However, as the completely hierarchy is illustrated within the first index of the partitions, the act of remapping is a simple string modification.
+If we want to remap a camera image from `camera1` to `camera2`, we have to modify the string in the respective field.
+Publishers or Subscribers are uneffected by this change and don't have to be shutdown and restarted.
+The way of modifying these strings, whether with complete string replacement or regex modifiers, exceeds the scope of this article.
+
+## Compare and Contrast with ROS 1
+
+In order to support a mapping to the more restrictive DDS topic name rules, these rules are in some ways more constraining than the rules for ROS 1.
+Other changes have been proposed for convenience or to remove a point of confusion that existed in ROS 1.
+In ROS 2, topic and service names differ from ROS 1 in that they:
+
+- must separate the tilde (`~`) from the rest of the name with a forward slash (`/`)
+
+  - This is done to avoid inconsistency with how `~foo` works in filesystem paths versus when used in a ROS name.
+
+- may contain substitutions which are delimited with balanced curly braces (`{}`)
+
+  - This is a more generic extension of the idea behind the tilde (`~`).
+
+- have length limits
+
+  - This is driven by the topic name length limit of DDS.
+
+- may be indicated as "hidden" by using a leading underscore (`_`) in one of the namespaces
+
+  - This is used to hide common but infrequently introspected topics and services.
+
+## Concerns/Alternatives
+
+This section lists concerns about the proposed design and alternatives that were considered.
 
 ### ROS Specific Name Prefix
 
+Initially, the idea of implementing ROS topics and namespaces based on specific name prefixes was chosen.
+This is not any longer the favorite solution as DDS partitions have proven to be more flexible and accurate since they are part of the protocol.
```

Comment:
> wasn't the point of the ROS prefix to help tools that wanted to ignore non-ROS DDS topics?

It was the rationale for it. Also, so that it is easy to answer the question "which of the DDS topics are also ROS topics?"

I don't know why @Karsten1987 arrived at that statement. I can only guess he didn't see a need for it or confused it's purpose as some mechanism specific for the substitution method proposed before.

I still think it has merit. If it is removed it should at least be covered by the alternatives (which currently cover why we *did* keep the prefix):

> This alternative would:

> - not prefix topics
> - optionally prefix other kinds of "implementation detail topics"

> Trade-offs:

> - it would be easier to have ROS subscribe to a DDS created topic

>   - e.g. DDS publisher on topic image could be subscribed to in ROS using just image
>   - however, the types would need to match as well
>   - in the current proposal the ROS topic image would become rt__image, so DDS topics would need to follow the ROS topic conversion scheme to interoperate with ROS components
>   - it would be hard to distinguish ROS created DDS topics and normal DDS topics

> - services would still need to be differentiated

>   - e.g. service /foo would need to make two topics, something like foo_Request and foo_Reply
Rationale:

> Slight preference was given to easily categorizing ROS created topics with DDS created topics over easily connecting to existing DDS topics. Connecting to DDS topics could be achieved by having an option when subscribing or publishing to "alias" to an implementation topic name, e.g. something like sub->alias_to_explicit_topic('dds_topic'). Also, the workaround for separating ROS created topics from other DDS topics was considered to be more complicated than the suggested solution of allowing users to specify specific DDS topic names for their publishers and subscriptions.
> -- https://github.com/ros2/design/blob/gh-pages/articles/140_topic_and_service_name_mapping.md#limited-prefixes-alternative

This section hasn't been changed in this pr (yet): https://github.com/ros2/design/pull/109/files#diff-51ea0dd0254ef288483a09be8b28379dR403

> if so, how have things changed by using partitions?

As far as I can tell the partitions don't have anything to do with the decision to implicitly namespace ROS topics so they stand apart from other DDS topics. I think they could be made independently. The only thing I can think of is that one of the arguments against the prefix (and why it was so short) was that it cut into the 255 topic length. The partitions alleviate that slightly since the combine namespace + topic name limit is now longer.